### PR TITLE
Test3: TicTacToe Game

### PR DESCRIPTION
**Pull request by DhaaraniMohanasundaram**

### CODE REVIEW

File: Training24/TicTacToe.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+         (board[0][i] == player && board[1][i] == player && board[2][i] == player))
+         return WIN;
+   }
+   if ((board[0][0] == player && board[1][1] == player && board[2][2] == player) ||
+      (board[0][2] == player && board[1][1] == player && board[2][0] == player))
+      return WIN;
+   for (int i = 0; i < BOARD_SIZE; i++) {
+      for (int j = 0; j < BOARD_SIZE; j++) if (board[i][j] == ' ') return CONTINUE;
+   }
+   return DRAW;
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c:55]: Add necessary comments

File: Training24/TicTacToe.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      if ((board[i][0] == player && board[i][1] == player && board[i][2] == player) ||
+         (board[0][i] == player && board[1][i] == player && board[2][i] == player))
+         return WIN;
+   }
+   if ((board[0][0] == player && board[1][1] == player && board[2][2] == player) ||
+      (board[0][2] == player && board[1][1] == player && board[2][0] == player))
+      return WIN;
+   for (int i = 0; i < BOARD_SIZE; i++) {
+      for (int j = 0; j < BOARD_SIZE; j++) if (board[i][j] == ' ') return CONTINUE;
+   }
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c]: Remove braces

File: Training24/TicTacToe.c [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+   return DRAW;
+}
+
+/// <summary>Gets a valid move from the player.</summary> 
+int GetValidMove (char board[BOARD_SIZE][BOARD_SIZE], char player) {
+   char input[10];
+   int move;
+   while (1) {
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c]: Instead of 1 use `while (true)` .

File: Training24/TicTacToe.c [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      if (gameStatus == WIN) {
+         PrintBoard (board);
+         printf ("Player %c wins!\n", currentPlayer);
+         break;
+      }
+      if (gameStatus == DRAW) {
+         PrintBoard (board);
+         printf ("It's a draw!\n");
+         break;
+      }
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c:93]: you can store both the board state and the messages in strings, and print them once at the end of the function

File: Training24/TicTacToe.c (hunk 2 above)
Comment (gdineshkumar219) [inline Training24/TicTacToe.c]: Add necessary comments for each block

File: Training24/TicTacToe.c [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+   while (1) {
+      printf ("\nPlayer %c, enter a move (1-9): ", player);
+      fgets (input, sizeof (input), stdin);
+      if (input[0] < '1' || input[0] > '9' || input[1] != '\0' && input[1] != '\n') {
+         printf ("Invalid input! Please enter a number between 1 and 9.\n");
+         continue;
+      }
+      move = input[0] - '0';
+      int row = (move - 1) / BOARD_SIZE,
+         col = (move - 1) % BOARD_SIZE;
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c]: Make this as single line

File: Training24/TicTacToe.c [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+      printf ("That spot is already taken! Please choose another spot.\n");
+   }
+}
+
+/// <summary>Starts two player TicTacToe game.</summary>
+void StartGame () {
+   char board[BOARD_SIZE][BOARD_SIZE] = { {' ', ' ', ' '}, {' ', ' ', ' '}, {' ', ' ', ' '} },
+      currentPlayer = 'X';
+   int move, gameStatus;
+   while (1) {
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c]: Instead of 1 use `while (true) `

File: Training24/TicTacToe.c [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+      currentPlayer = (currentPlayer == 'X') ? 'O' : 'X';   // Switch to the other player
+   }
+   PrintBoard (board);
+   printf ("%s\n", gameMessage);
+}
+
+int main () {
+   StartGame ();
+   return 0;
+}
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c:101]: image
1.In this first fill the board with numbers starting from 1 to 9  
image
2.These are all valid inputs why its showing it as invalid ? Pls check
3. Use clear screen to print only the reference board and updated board in the console.

File: Training24/TicTacToe.c [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+      board[row][col] = currentPlayer;
+      gameStatus = CheckGameStatus (board, currentPlayer);
+      if (gameStatus == WIN) {
+         snprintf (gameMessage, sizeof (gameMessage), "Player %c wins!", currentPlayer);
+         break;  //  End the game
+      }
+      if (gameStatus == DRAW) {
+         snprintf (gameMessage, sizeof (gameMessage), "It's a draw!");
+         break;
+      }
```
Comment (gdineshkumar219) [inline Training24/TicTacToe.c:93]: ` if (gameStatus != CONTINUE) {
            snprintf(gameMessage, sizeof(gameMessage), gameStatus == WIN ? "Player %c wins!" : "It's a draw!", currentPlayer);
            break;
        }` Try to use ternary wherever it is possible.

**Review by Dakshita-Minocha — APPROVED** [review]

**Pull request closed without merging**